### PR TITLE
QNTM-1709 Sync Node schema with Latest CoGs schema

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

[QNTM-1709](https://jira.autodesk.com/browse/QNTM-1709) Sync Node schema with Latest CoGs schema

According to @gregmarr 's proposal:
1. The `NodeType` for `Dynamo.Graph.Nodes.CustomNodes.Function, DynamoCore` becomes `FunctionNode` instead of `CustomFunctionNode`.
2. `FunctionUuid` is changed to `FunctionSignature`
3. `CustomNodes.Function` writes `"FunctionType": "Graph"` in the JSON.
4. `Dynamo.Graph.Nodes.ZeroTouch.DSVarArgFunction, DynamoCore` writes `"FunctionType": "VariableArguments"` in the JSON.

```json
"Nodes": [
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CustomNodes.Function, DynamoCore",
      "FunctionSignature": "4a21835f-29e8-4d8d-8130-0a00be7191fe",
      "FunctionType": "Graph",
      "NodeType": "FunctionNode",
      "Id": "14982afe204345f1b5cbb2298d41dcd5",
      "Inputs": [
        {
          "Id": "d3a8a324ad6245179e90a0a0a96d8685",
          "Name": "x",
          "Description": "int\nDefault value : 4 (disabled)",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "d5fd48ea29124f2495e3415180739753",
          "Name": "",
          "Description": "return value",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "1111"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSVarArgFunction, DynamoCore",
      "FunctionSignature": "DSCore.String.Join@string,string[]",
      "FunctionType": "VariableArguments",
      "NodeType": "FunctionNode",
      "Id": "89dc53c00d954c49b06b06cb40789641",
      "Inputs": [
        {
          "Id": "be7a90c18e5d492fbf6b16f094769cf4",
          "Name": "separator",
          "Description": "String to be inserted between joined strings.\n\nstring",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "b53eb61f0acf46d19d969da749de0c08",
          "Name": "string0",
          "Description": "Strings to be joined into a single string.\n\nstring[]",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "bbefa9274fbe43efa11c460baedc5961",
          "Name": "str",
          "Description": "A string made from the list of strings including the separator character.",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Concatenates multiple strings into a single string, inserting the given separator between each joined string.\n\nString.Join (separator: string, strings: string[]): string"
    }
  ]
```

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@ramramps @gregmarr 

### FYIs

@ColinDayOrg @alfarok

### CODE REVIEW

File: src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs [hunk 1]
```diff
@@ -74,19 +74,27 @@ public void UpdatePortsForUnresolved(PortModel[] inputs, PortModel[] outputs)
         /// <summary>
         /// The unique id of the underlying function.
         /// </summary>
-        public Guid FunctionUuid
+        public Guid FunctionSignature
         {
             get { return Definition.FunctionId; }
         }
 
+        public string FunctionType
```
Comment (ramramps) [inline src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: add some comments.

Comment (QilongTang) [reply src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: @gregmarr Can you speak to what is this `FunctionType` used for?

Comment (gregmarr) [reply src/DynamoCore/Graph/Nodes/CustomNodes/Function.cs]: It indicates which of the three types of function calls this node represents, a call to an external graph, a call to a function with a vararg argument, or a standard function.

**Comment by QilongTang** [comment]
@ramramps Any more comments on this one?

**Comment by ramramps** [comment]
@QilongTang LGTM

**Pull request merged**